source/develop/release-management/features/network/lldp.html.md

```diff
@@ -0,0 +1,466 @@
+---
+title: Link Layer Discovery Protocol (LLDP) Support
+category: feature
+authors: Dominik Holler
+wiki_category: Feature
+wiki_title: Link Layer Discovery Protocol (LLDP) Support
+wiki_revision_count: 1
+wiki_last_updated: 2016-06-27
+feature_name: Link Layer Discovery Protocol (LLDP) Support
+feature_modules: engine,network,vdsm,rest-api
+feature_status: In Development
+---
+
+# Link Layer Discovery Protocol (LLDP) Support
+
+## Summary
+
+The [Link Layer Discovery Protocol (LLDP)][1] is a vendor-neutral networking
+protocol, used by network devices, especially switches, for advertising their
+identity, capabilities and management addresses on an IEEE 802 local area
+network (LAN).
+
+The feature will enable oVirt hosts to gather information from their networking
+interfaces and let oVirt Engine use this information in a beneficial way.
+
+## Owner
+
+*   Name: [Dominik Holler](https://github.com/dominikholler)
+
+*   Email: <dholler@redhat.com>
+
+## Detailed Description
+
+To support the integration of oVirt with other software, the information gathered
+by LLDP on each host will be provided by oVirt Engine's REST-API.
+
+Some of the information, like the number of the switch port a network interface
+of the host is connected to, might be helpful for the user and will be shown in
+the graphical user interface.
+
+At this point in time we do not intend to expose information by LLDP to the
+switch.
+
+## Prerequisites
+
+LLDP has to be enabled on the switch connected to the relevant oVirt hosts.
+
+## Limitations
+
+* There is a delay from receiving the LLDP data on the oVirt host, propagating
+  the data to the oVirt Engine and presenting the data to the user.
+  Whatever value we present over REST may be out of date.
+
+## Benefit to oVirt
+
+Information from LLDP provided by oVirt engine's REST-API will support the
+integration of oVirt in other software.
+
+Information from LLDP provided by the UI might help the user to avoid and
+locate manual host network configuration errors caused by a mismatch of switch
+port configuration.
+
+## Requirements
+
+* The following TLVs have to be available in oVirt Engine's REST API:
+  * Chassis ID TLV (Type = 1)
+  * Port ID TLV (Type = 2)
+  * Port Description TLV (Type = 4), with a value of up to 2000 chars
+  * System Name TLV (Type = 5)
+  * System Description TLV (Type = 6)
+  * Management Address TLV (Type = 8)
+  * Port VLAN Id ( Type = 127, Oui = 0x0080c2 (802.1), Subtype = 1)
+  * VLAN name (Type = 127, Oui = 0x0080c2 (802.1), Subtype = 3)
+
+## Open Questions
+
+* How is a Port Description with a length than more than 512 chars represented
+  in LLDP?
+* Should oVirt Engine provide data of hosts, which are currently not available
+  or non-operational?
+* Should oVirt Engine provide the age of the data?
```

Comment:
I ignored this comment, because the UI is not affected, and with the current approach only fresh data are provided.